src/Expander.hs

```diff
@@ -1400,12 +1415,17 @@ interpretMacroAction prob =
           phase <- view (expanderLocal . expanderPhase)
           s <- getState
           let env = fromMaybe Env.empty .
-                    view (expanderWorld . worldEnvironments . at phase) $
-                    s
-          value <- expandEval $ withEnv env $ apply closure boundResult
-          case value of
-            ValueMacroAction act -> interpretMacroAction prob act
-            other -> throwError $ ValueNotMacro other
+                    view (expanderWorld . worldEnvironments . at phase) $ s
+          case applyInEnv env closure boundResult of
```

Comment:
another place where this MR highlights a mistake in the original code: all of
```
phase <- view (expanderLocal . expanderPhase)
s <- getState
let env = ...
```
and the `withEnv env` is dead code, because `apply` overwrites this `env` with its own call to `withEnv`. So there is no need for an `applyInEnv` separate from `apply`.

Reply:
```suggestion
          phase <- view (expanderLocal . expanderPhase)
          case apply closure boundResult of
            Left err -> throwError
              $ MacroEvaluationError phase
              $ projectError err
```